src/components/AudioPlayer.tsx

```diff
@@ -0,0 +1,134 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {PlayerState} from '../store';
+import {Player} from 'tone';
+import Fader from './ui/Fader';
+
+export type PlayerController = (props: {player?: Player}) => PlayerState;
+
+export const usePlayerController: PlayerController = ({player}) => {
+  const [isPlaying, setIsPlaying] = useState(false);
+
+  const onPlay = () => {
+    player?.start();
+    setIsPlaying(true);
+  };
+
+  const onStop = () => {
+    player?.stop();
+    setIsPlaying(false);
+  };
+
+  const onRestart = (loopStart: number) => {
+    player?.restart(loopStart);
+  };
+
+  const setAttribute = (setState: {
+    playbackRate?: number;
+    loopStart?: number;
+    loopEnd?: number;
+  }) => {
+    player?.set({...player?.get(), ...setState});
+  };
+
+  return {isPlaying, onStop, onPlay, onRestart, setAttribute};
+};
+
+interface AudioPlayerProps {
+  isPlaying?: boolean;
+  onPlay?: () => void;
+  onStop?: () => void;
+  onRestart?: (value: number) => void;
+  onInputChange?: ({}) => void;
+}
+
+const AudioPlayer: React.FC<AudioPlayerProps> = ({
```

Comment:
bonus: can you write a test to test this function component?

Reply:
let's talk about this later